src/core/CMakeLists.txt

```diff
@@ -0,0 +1,84 @@
+find_package(LZ4)
+find_package(Boost COMPONENTS
+  system
+  program_options
+  REQUIRED)
+find_package(PkgConfig REQUIRED)
+smfc_gen(
+  CPP
+  TARGET_NAME raw_rpc_gen
+  OUTPUT_DIRECTORY ${PROJECT_SOURCE_DIR}/src/smf
```

Comment:
use binary directory to not pollute out of source tree builds

Reply:
good point!

Reply:
so for the core, it needs to be in smf/rpc_generated.h no ? - also, this is the exception in that other projects will need to check this in. I was actually thinking about manually checking this one in. what do you think ?

Reply:
if you just have 

`include` and `libsmf.a` then you need this generated in the `include/smf/* `

Reply:
added a comment in #221

Reply:
mmm

The way I would expect it to work would be to include `smf/rpc_generated.h` but `rpc_generated.h` will be in `<build path>/src/smf/`. targets will include the build path, and we'll have to figure out how to make target_include_directories not _remember_ that path for the version of the library that is installed. I think that's related to cmake rpath stuff...

1) this should be a non-issue for those using smf via cmake + add_subdirectory

2) for users of smf that have done `make install` or in the future from `dnf/apt-get`, `make install` should install `<build path>/src/smf/rpc_generated.h` to `<install path>/smf/`

checking in this file is definitely also an option if there is no risk that flatc generates code that is platform dependent. that might be an FAQ on the flatbuffers project.

Reply:
there isn't a whole lot of platform dependent code generated. Will generate it manually. THanks!

Also, do you mind If I make a copy of your repo at it's current stage and refer ppl to it to get started?

Reply:
yeh, feel free to share that project link. it's probably not a good example yet :)

why have a guard to regenerate rpc.fbs when smfc_gen will only regenerate when rpc.fbs  changes anyway? as a maintainer you could add it to .gitignore and use git add -f when rpc.fbs actually changes.

Reply:
that's a good point.